### ISSUE TITLE
wofi-1.2.1 crashes under sway-1.5.1

### ISSUE THREAD

**Issue by ddelabru** [NixOS/nixpkgs#103820]
**Describe the bug**
wofi immediately crashes upon invocation when sway is running as the window manager

**To Reproduce**
1. Start sway (`$ sway`)
2. Run wofi (e.g. `$ wofi --show run`). wofi exits with this message: `Gdk-Message: 11:36:41.128: Error 71 (Protocol error) dispatching to Wayland display.`

**Expected behavior**
wofi should display a special floating application launcher window.

**Additional context**
According to [a now-closed issue filed against sway](https://github.com/swaywm/sway/issues/5803), this problem has been fixed on the main SCM branch of wofi

I am using the unstable NixOS channel. Some versions:
* wofi: 1.2.1
* sway: 1.5.1
* gdk-pixbuf (as a wofi dependency): 2.40.0

One work-around is to invoke wofi with the `-n` or `--normal-window` flag. This works for me, but was not necessary with other versions and is less than ideal because it displays wofi as a normal window with different behavior than what wofi would otherwise have; for example by default this window is drawn with the same window border and title bar that any non-launcher floating window would have.

**Notify maintainers**
@Elyhaka is listed as the wofi maintainer.

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.77, NixOS, 21.03pre251971.a371c107116 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.8`
 - channels(root): `"nixos-21.03pre251971.a371c107116"`
 - channels(lunasspecto): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- wofi
# a list of nixos modules affected by the problem
module: []
```

**Comment by Technical27** [comment]
already fixed in #103701

you can use this temporarily as an overlay
```nix
      wofi = super.wofi.overrideAttrs (old: rec {
        version = "1.2.3";
        src = super.fetchhg {
          url = "https://hg.sr.ht/~scoopta/wofi";
          rev = "v${version}";
          sha256 = "0glpb2gf5n78s01z3rn614ak8ibxhfr824gy6xlljbxclgds264i";
        };
      });
```

**Comment by ddelabru** [comment]
Now that the wofi fix is in nixos-unstable I can confirm it's fixed and am closing this issue, thanks!
